plugin/vimux.vim

```diff
@@ -3,20 +3,26 @@ if exists('g:loaded_vimux') || &compatible
 endif
 let g:loaded_vimux = 1
 
-function! s:VimuxOption(option, default)
-  if exists(a:option)
-    return eval(a:option)
-  else
-    return a:default
-  endif
-endfunction
+" Set up all global options with defaults right away, in one place
+let g:VimuxDebug         = get(g:, 'VimuxDebug',         v:false)
+let g:VimuxHeight        = get(g:, 'VimuxHeight',        20)
+let g:VimuxOpenExtraArgs = get(g:, 'VimuxOpenExtraArgs', '')
+let g:VimuxOrientation   = get(g:, 'VimuxOrientation',   'v')
+let g:VimuxPromptString  = get(g:, 'VimuxPromptString',  'Command? ')
+let g:VimuxResetSequence = get(g:, 'VimuxResetSequence', 'q C-u')
+let g:VimuxRunnerName    = get(g:, 'VimuxRunnerName',    '')
+let g:VimuxRunnerType    = get(g:, 'VimuxRunnerType',    'pane')
+let g:VimuxTmuxCommand   = get(g:, 'VimuxTmuxCommand',   'tmux')
+let g:VimuxUseNearest    = get(g:, 'VimuxUseNearest',    v:true)
+
 
-function! s:VimuxTmuxCmd()
-  return s:VimuxOption('g:VimuxTmuxCommand', 'tmux')
+function! s:VimuxOption(name) abort
+  return get(b:, a:name, get(g:, a:name))
```

Comment:
This is nice, I like the buffer first, otherwise global scope and the fact that it happens on use.

This extended feature should be documented somewhere in the help docs so people know they can set buffer scope vars without messing up the user's global defaults. This should come in handy for a number of the language specific add-ons out there for this plugin.

Reply:
Good point. I don't have time today but I'll add some documentation with this PR when I have a chance.

Reply:
I already added a note in the documentation.